mypy/semanal.py

```diff
@@ -1052,14 +1052,15 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
             allow_tuple_literal = isinstance(s.lvalues[-1], (TupleExpr, ListExpr))
             s.type = self.anal_type(s.type, allow_tuple_literal)
         else:
-            # For simple assignments, allow binding type aliases.
+            # For simple assignments at top level, allow binding type aliases.
             if (s.type is None and len(s.lvalues) == 1 and
-                    isinstance(s.lvalues[0], NameExpr)):
+                    isinstance(s.lvalues[0], NameExpr) and
+                    not self.is_func_scope() and not self.type):
                 res = analyze_type_alias(s.rvalue,
                                          self.lookup_qualified,
                                          self.lookup_fully_qualified,
                                          self.fail)
-                if res and (not isinstance(res, Instance) or cast(Instance, res).args):
+                if res:
```

Comment:
IIUC the code you removed here was added in ab4a0fee25256a52f421903e2990de41672e7d13 and while some of the surrounding code was changed, it seems to guard against using e.g. `List[AnyStr]`. Can you check that either that's still prohibited or it actually works as expected?